nexus/reconfigurator/execution/src/dns.rs

```diff
@@ -1045,7 +1049,33 @@ mod test {
             String::from("oxide.test"),
         );
         assert_eq!(external_dns_zone.zone_name, "oxide.test");
-        assert!(external_dns_zone.records.is_empty());
+        // We'll only have external DNS nameserver records - the A/AAAA records
+        // for servers themselves, and NS records at the apex.
+        let baseline_external_dns_names = external_dns_count + 1;
+        assert_eq!(
+            external_dns_zone.records.len(),
+            baseline_external_dns_names
+        );
```

Comment:
If I'm following this right, this is asserting that there are 4 external DNS zone records.  I don't follow why that's right, though -- don't we have an A record and NS record for each of the 3 servers, so 6 records altogether?

Edit: Oh, is `records.len()` the number of different *names* that have any records at all?  So it's: the apex (which has the 3 NS records all at one name) + each of the three servers (each with a different name)?  If that's it maybe the comment could be a little more explicit.  "Although there will be 2 records for each external DNS server, all the NS records will be under one name, so there are only N + 1 entries in the `records` map."

Reply:
i was pretty confused at first here too, for the same reason. i'd tried writing exactly that `assert_eq!(...len(), 6)` even! i'll change `.records` to `.names` in the v2 type and i think that will make it more legible.

Reply:
https://github.com/oxidecomputer/omicron/pull/8047/commits/ff63ea15350eace1f7ce45db981088f192e697a9 as i was going through this there are several other cases where getting a `records` from a `records` seemed confusing where picking a name and records out of a `records` reads much better. so, seems good.